### PR TITLE
Switch from go-get to go-install and bump linter version

### PR DESCRIPTION
**Pull request by raypinto**
**Description**

Changes proposed in this pull request:

- Remove deprecated `go get` and use `go install` instead
- Bump golangci-lint to 1.45.2 as go1.18 is supported from v1.45.0.

**Related issue(s)**

### CODE REVIEW

File: components/eventing-controller/Makefile [hunk 1]
```diff
@@ -92,8 +92,9 @@ undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/confi
 	kustomize build config/default | kubectl delete -f -
 
 CONTROLLER_GEN = $(PROJECT_DIR)/bin/$(shell uname)/controller-gen
+ARCH := $(shell uname)
```
Comment (k15r) [inline components/eventing-controller/Makefile:95]: i know this was in there before, but can we rename it from ARCH to OS as it is not the architecture, but the os we are interested in

Comment (raypinto) [reply components/eventing-controller/Makefile]: done

**Review by clebs — CHANGES_REQUESTED** [review]
Please replace the deprecated linters:
golint -> revive
scopelint -> exportloopref

**Review by clebs — APPROVED** [review]

**Review by k15r — APPROVED** [review]

**Pull request merged**